src/block.c

```diff
@@ -730,6 +730,29 @@ coap_add_data_large_response(coap_resource_t *resource,
   return 0;
 }
 
+coap_tick_t
+coap_block_check_lg_xmit_timeouts(coap_session_t *session, coap_tick_t now) {
+  coap_lg_xmit_t *p;
+  coap_lg_xmit_t *q;
+  coap_tick_t partial_timeout = 8 * COAP_TICKS_PER_SECOND;
+  coap_tick_t tim_rem = -1;
+
+  LL_FOREACH_SAFE(session->lg_xmit, p, q) {
+    if (p->last_used &&
+        p->last_used + partial_timeout <= now) {
+      /* Expire this entry */
+      LL_DELETE(session->lg_xmit, p);
+      coap_block_delete_lg_xmit(session, p);
+    }
+    else if (p->last_used) {
```

Comment:
For better readability, I suggest restructuring the if-clause to check for `p->last_used` and then check for the timeout:

```c
if (p->last_used == 0) {
  continue;
}

if (p->last_used + partial_timeout <= now) {
   ...
} else {
  if (tim_rem > ...) ....
}
```

Reply:
Code updated as suggested and code pushed.